src/nbl/video/COpenGL_Swapchain.h

```diff
@@ -253,10 +279,11 @@ class COpenGL_Swapchain final : public ISwapchain
             }
 
             gl.extGlBlitNamedFramebuffer(fbos[imgix], 0, 0, 0, w, h, 0, 0, w, h, GL_COLOR_BUFFER_BIT, GL_NEAREST);
-            egl->call.peglSwapBuffers(egl->display, surface);
             syncs[imgix] = core::make_smart_refctd_ptr<COpenGLSync>();
             syncs[imgix]->init(m_device, &gl, false);
-            gl.glGeneral.pglFlush();
+            // swap buffers performs an implicit flush before swapping 
+            // https://www.khronos.org/registry/EGL/sdk/docs/man/html/eglSwapBuffers.xhtml
```

Comment:
maybe we have to emulate this? Do WGL and GLX swapbuffers perform implicit flush as well?

Reply:
if they don't, then its the responsibility of the `libEGL` to ensure they do.

Reply:
theres no libEGL XD its our implementation

Reply:
yeah I mean check this implicit flush on GLX and WGL, adjust in our implementation as necessary ;)

Reply:
resolved?